manager/integration/tests/test_node.py

```diff
@@ -1045,3 +1084,79 @@ def test_replica_cleanup(client):  # NOQA
                                        len(update_disks))
 
     cleanup_host_disk(client, 'extra-disk')
+
+
+@pytest.mark.node
+def test_node_default_disk_labeled(client, core_api, disk_path,  reset_default_disk_label,  # NOQA
+                                   reset_disk_settings):  # NOQA
+    """
+    Test that only Nodes with the proper label applied get a default Disk
+    created on them when one doesn't already exist. Makes sure the created
+    Disk matches the Default Data Path Setting.
+    """
+    # Set up cases.
+    cases = {
+        "disk_exists": None,
+        "labeled": None,
+        "unlabeled": None
+    }
+    nodes = client.list_node()
+    assert len(nodes) >= 3
+
+    node = nodes[0]
+    cases["disk_exists"] = node["id"]
+    core_api.patch_node(node["id"], {
+        "metadata": {
+            "labels": {
+                CREATE_DEFAULT_DISK_LABEL: "true"
+            }
+        }
+    })
+
+    node = nodes[1]
+    cases["labeled"] = node["id"]
+    core_api.patch_node(node["id"], {
+        "metadata": {
+            "labels": {
+                CREATE_DEFAULT_DISK_LABEL: "true"
+            }
+        }
+    })
+    disks = node["disks"]
+    for _, disk in disks.iteritems():
+        disk["allowScheduling"] = False
+    update_disks = get_update_disks(disks)
+    node = node.diskUpdate(disks=update_disks)
+    node = node.diskUpdate(disks=[])
+    wait_for_disk_update(client, node["id"], 0)
+
+    node = nodes[2]
+    cases["unlabeled"] = node["id"]
+    disks = node["disks"]
+    for _, disk in disks.iteritems():
+        disk["allowScheduling"] = False
+    update_disks = get_update_disks(disks)
+    node = node.diskUpdate(disks=update_disks)
+    node = node.diskUpdate(disks=[])
+    wait_for_disk_update(client, node["id"], 0)
+
+    # Set disk creation and path Settings.
+    setting = client.by_id_setting(CREATE_DEFAULT_DISK_SETTING)
+    client.update(setting, value="true")
+    setting = client.by_id_setting(DEFAULT_DATA_PATH_SETTING)
```

Comment:
I think you need to update DEFAULT_DATA_PATH_SETTING first, otherwise, the creation can happen as soon as you change the CREATE_DEFAULT_DISK_SETTING to true.